libs/journeys/ui/src/libs/EditorProvider/EditorProvider.tsx

```diff
@@ -33,6 +33,7 @@ export enum ActiveJourneyEditContent {
 export interface EditorState {
   steps?: Array<TreeBlock<StepBlock>>
   selectedStep?: TreeBlock<StepBlock>
+  selectedComponent?: string
   selectedBlock?: TreeBlock
```

Comment:
Thoughts on this? Since either a block or the footer can be selected?
```suggestion
  selectedBlock?: TreeBlock || 'Footer'
```

Reply:
It's much more complex to change selectBlock to have multiple types than to create a new state prop. We will have to implement type-checks everywhere we use "selectedBlock". 

In addition they want to support editing of other non-block elements like the step header in the future so kept it seperate.